packages/x/x264/xmake.lua

```diff
@@ -59,7 +66,27 @@ package("x264")
                 end
             end
         end
-        import("package.tools.autoconf").install(package, configs)
+        if package:is_plat("windows") then
+            import("core.base.option")
+            import("core.tool.toolchain")
+            local msvc = toolchain.load("msvc", {plat = package:plat(), arch = package:arch()})
+            assert(msvc:check(), "msvs not found!")
+            local envs = os.joinenvs(os.getenvs(), msvc:runenvs()) -- keep msys2 envs in front to prevent conflict with possibly installed sh.exe
+            envs.CC = path.filename(package:build_getenv("cc"))
+            envs.SHELL = "sh"
+
+            table.insert(configs, "--toolchain=msvc")
+            table.insert(configs, "--prefix=" .. package:installdir())
+            os.vrunv("sh", table.join("./configure", configs), {envs = envs})
```

Comment:
`os.vrunv("./configure", configs, {shell = true})` will call `sh` to load configure.